targets/TARGET_TOSHIBA/TARGET_TMPM3HQ/rtc_api.c

```diff
@@ -0,0 +1,220 @@
+ * limitations under the License.
+ */
+#include "rtc_api.h"
+#include "mbed_mktime.h"
+
+#define RTC_24_HOUR_MODE                ((uint8_t)0x01)
+#define PAGER_PAGE_ONE                  ((uint8_t)0x01)
+#define PAGER_PAGE_ZERO                 ((uint8_t)0xEE)
+#define RTC_CLK_ENABLE                  ((uint8_t)0x08)
+#define RTC_CLK_DISABLE                 ((uint8_t)0xE7)
+#define RTCRESTR_RSTTMR_MASK            ((uint8_t)0x20)
+#define RTCRESTR_RSTTMR_R_RUN           ((uint8_t)0x20)
+#define CGWUPLCR_WUPTL_HIGH_MASK        ((uint32_t)0x07FFF000)
+#define CGWUPLCR_WULEF_MASK             ((uint32_t)0x00000002)
+#define CGWUPLCR_WULEF_R_DONE           ((uint32_t)0x00000000)
+#define CGWUPLCR_WULON_W_ENABLE         ((uint32_t)0x00000001)
+#define RLMLOSCCR_XTEN_RW_ENABLE        ((uint32_t)0x00000001)
+#define ELOSC_CFG_WARM_UP_TIME          ((uint64_t)(5000))
+#define ELOSC_CFG_CLOCK                 ((uint64_t)(32768))
+#define HEX2DEC(val)                    ((val >> 4U) * 10U + val % 16U)  // Hex to Dec conversion macro
+#define DEC2HEX(val)                    ((val / 10U) * 16U + val % 10U)  // Dec to Hex conversion macro
+
+static int flag = 0;
+static int diff_year = 100;         //our RTC register only support 2000~2099
+static void external_losc_enable(void);
+
+void rtc_init(void)
+{
+    if (!flag) {
+        TSB_CG_FSYSENA_IPENA23 = 1;  // Enable Sys Clock for RTC
+        external_losc_enable();      // Enable low-speed oscillator
+
+        TSB_RTC->PAGER = 0x00;       //disable clock and alarm
+
+        while ((TSB_RTC->RESTR & RTCRESTR_RSTTMR_MASK) == RTCRESTR_RSTTMR_R_RUN) {
+            // Reset RTC sec counter
+        }
+
+        TSB_RTC->RESTR = 0xE7;
+
+        while ((TSB_RTC->RESTR & RTCRESTR_RSTTMR_MASK) == RTCRESTR_RSTTMR_R_RUN) {
+            // Reset RTC sec counter
+        }
+
+        TSB_RTC->PAGER   |= PAGER_PAGE_ONE;
+        TSB_RTC->YEARR   = 0x03;              // Set leap year state
+        TSB_RTC->MONTHR  = RTC_24_HOUR_MODE;  // Set hour mode
+        TSB_RTC->PAGER  &= PAGER_PAGE_ZERO;   // Set hour mode
+        TSB_RTC->YEARR   = 0x01;              // Set year value
+        TSB_RTC->MONTHR  = (uint8_t)0x01;     // Set month value
+        TSB_RTC->DATER   = (uint8_t)0x01;     // Set date value
+        TSB_RTC->DAYR    = (uint8_t)0x0;      // Set day value
+        TSB_RTC->HOURR   = (uint8_t)0x01;     // Set hour value
+        TSB_RTC->MINR    = (uint8_t)0x02;     // Set minute value
+        TSB_RTC->SECR    = (uint8_t)0x22;     // Set second value
+        TSB_RTC->PAGER  |= RTC_CLK_ENABLE;    // Enable Clock
+        flag             = 1;                 // Enable internal flag
+    }
+}
+
+void rtc_free(void)
+{
+    if (flag) {  // Check status of RTC peripheral driver is ENABLE or DISABLE
+        flag = 0;  // Set status of RTC peripheral driver is DISABLE
+    }
+}
+
+int rtc_isenabled(void)
+{
+    return flag;  // Return a flag that represents status of RTC peripheral driver
```

Comment:
same question as in another PR if this is usable after app resets

Reply:
@0xc0170, RTC status register is not in Toshiba MCUs. However, verified with RTC initialized status.

Reply:
https://github.com/ARMmbed/mbed-os/pull/10874#issuecomment-508735622